msteams-platform/concepts/design/understand-use-cases.md

```diff
@@ -5,47 +5,84 @@ description: Understand your use cases
+## Understand the limitations of the app
 
-Sometimes the answer is "We want to solve this problem for all Team's users everywhere." If that's the case for you you'll want to spend some time understanding [what it takes to get published to AppSource](~/concepts/deploy-and-publish/appsource/prepare/submission-checklist.md).
+Knowing the limitations of the apps in terms of data accessibility and data residency requirement will help you design better apps. This is important, as having information on who owns the data and availability of APIs impacts the solution architecture. Again, ask relevant questions, such as:
 
-## Do you need authentication?
+* What are the challenges with back end integration of the current app?
+* Who owns the back end data? In-house or third-party.
+* Are there firewalls that impact the functioning of the app?
+* Does it have APIs? 
 
-You should identify early on if you're going to need to protect the services you're exposing, and at what level. Remember the web services you'll be exposing in your Teams app are publicly available over the internet, so if you need to secure them start thinking about how now.
+## Provide authentication
 
-## Should the entire app be in Teams?
+You must identify early on if you need to protect the services you are exposing and at what level. Remember, the web services exposed in your Teams app are publicly available over the internet. So, if you need to secure them start thinking about it now. If you need a solution that requires you to provide guest access for users outside the tenant, access restrictions and permissions need to be placed to protect confidential information. You will need to design apps considering the limitations that come with guest user access. Therefore, ask questions, such as: 
 
-Whether you're building something entirely new, or bringing an existing solution into Teams, it is important to decide if the entire app is going to be inside the Teams client, or if it makes sense to only bring in a portion of the experience. With a combination of tabs, messaging extensions, task modules, interactive cards, and conversational bots you can build complex apps completely inside of Teams. However, that doesn't always make sense. Remember who your users are, and the problem you're trying to solve. Do they already have a system for solving most of the problem, and you just need to extend a sub-set of the functionality into Teams? Typically if you're only going to bring in a portion of your solution you should focus on sharing, collaboration, and initiating and monitoring workflows.
+* Will the users access different views of data based on their roles?
+* Is there PII involved?
+* Will the interactions also be based on the user roles?
+* Will external users access the app?
 
-## What will the onboarding experience be like?
+## Build entire app within Teams
```

Comment:
```suggestion
## Decide what goes in Teams
```
@KirtiPereira Suggested edit to reflect the text. We cannot state everyone should include their entire app in Teams. Some devs only want a couple key features in Teams, for instance.